lebedigital/raw_data_processing/metadata_extraction/mixture_metadata_extraction.py

```diff
@@ -1,142 +1,190 @@
+# 3.a Look for all sheets containing the word "Rezeptur" and load them
+# 3.b For each sheet: find indices of the labels
+# 3.c For each sheet: create and fill a dic & export dic as yaml-file
 
-# In[1]: Setup
+#------------------------------------------------------------------------------
 
 from cmath import nan
 import pandas as pd
 import os
 import yaml
 
-def translate(
-        df, 
-        dic
-        ):
-
-    """
-        Translates german labels to english according to the translation.xls
-
-        Parameter
-        ---------
-        df
-            The dataframe with the german labels that should be translated.
-        dic
-            The excelfile that contains the translations, columns are "german" and "english"
-    """
-    
-    translation = pd.read_excel(dic)
-    translation.columns = ["german", "english"]
-
-    for deutsch, english in translation.itertuples(index=False,name=None):
-        df = df.replace(deutsch,english,regex=True)
-    return df
+# function to convert german formatting to english
+def replace_comma(string):
+    if "---" in string:
+        string = nan
+        return string
+    else:
+        string = string.replace(',', '.')
+        return float(string)
 
 
-
-def extraction(
+# extraction script
+def extract_metadata_mixture(
         locationOfRawData,
-        locationOfProcessedData,
-        english = True
+        locationOfProcessedData
         ):    
 
     """
-        Extractes the metadata from the "Rezeptur"-Sheet of a given datafile (xls or xlsx).
+        Extracts the metadata from all "Rezeptur"-sheets of a given datafile 
+        (xls or xlsx). Creates one yaml-file per sheet containing the keyword
+        "Rezeptur".
+
 
         Parameter
         ---------
-        locationOfRawData
-            Path of the excelsheet (xls or xlsx) containing the metadata in one or multiple "Rezeptur"-Sheet(s).
-        locationOfProcessedData
+        locationOfRawData : string
+            Path of the excelsheet (xls or xlsx) containing the metadata in one 
+            or multiple "Rezeptur"-Sheet(s).
+        locationOfProcessedData : string
             Path of the target folder for yaml-file.
-        english
-            Boolean, checking if the metadata labels should be english or german 
-            -> For True, it uses a provided dictionary. 
-        
+
+        Output
+        -------
+        yamlFile containing the dictionary with the metadata
+                    
     """
     
+
+    # Find sheets in the file containing the mixture (keyword: "Rezeptur")
     excelsheet = os.path.basename(locationOfRawData)
     print("\n Working on: "+ excelsheet)
     excelfile = pd.read_excel(locationOfRawData, sheet_name= None) 
     listofkeys = [i for i in excelfile.keys() if "Rezeptur" in i] 
-    print("Relevant sheets in this file: " + str(listofkeys))
+    print("Sheets containing mixture metadata in this file: " + str(listofkeys))
```

Comment:
I think we should add a logger (e.g. loguru as used [here](https://github.com/BAMresearch/probeye/blob/6272085e8c69131b43b6a90c81f502636d3aa59d/probeye/definition/forward_model.py#L5) not to clutter the output with debugging information. An simple alternative would be to define a debug=True variable at the top and put that print statements into an if statement.